hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/EthereumTransactionHandler.java

```diff
@@ -179,6 +180,12 @@ public void handleThrottled(@NonNull final HandleContext context) {
     @Override
     public Fees calculateFees(@NonNull final FeeContext feeContext) {
         requireNonNull(feeContext);
+        final var zeroHapiFeeEnabled =
+                feeContext.configuration().getConfigData(ContractsConfig.class).evmEthTransactionZeroHapiFeesEnabled();
+        return zeroHapiFeeEnabled ? Fees.FREE : getLegacyCalculateFees(feeContext);
```

Comment:
I think it's unfortunate we can't use the "proper" designed fee mechanism because it doesn't handle different costs for success vs failed. 

Anyway, this line with `Fees.FREE` make it clear the fee isn't charged _here_ (when the config option is set).  Consider a comment here pointing off to `ContextTransactionProcessor` esp. the method `chargeOnFailedEthTxn` which is where the fee is charged in case of failure.

Reply:
Added Javadoc explaining the situation and pointing to the method. Thanks!